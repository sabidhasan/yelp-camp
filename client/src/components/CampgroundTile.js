import React from 'react'
import RatingBar from './RatingBar'

const CampgroundTile = (props) => {
  return (
    <React.Fragment>
      <img src={props.image} alt="" />
      <a href={'campground/' + props.id}>
        <h2>{props.name}</h2>
      </a>
      <span className='region'>{props.region}, {props.province}</span>
      <div className='rating'>
        <RatingBar rating={props.rating} small={true} />
      </div>
      <span className='review-count'>
        {props.ratingCount ? `${props.ratingCount} Reviews` : 'No reviews yet' }
      </span>
    </React.Fragment>
  )
}

export default CampgroundTile
