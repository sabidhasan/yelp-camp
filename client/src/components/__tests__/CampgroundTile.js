import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import toJson from 'enzyme-to-json'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import CampgroundTile from '../CampgroundTile'

configure({ adapter: new Adapter() });

describe('<CampgroundTile />', () => {
  let component;
  let props;
  const image = 'image_url'
  const signIn = jest.fn();

  beforeEach(() => {
    props = {
      ratingCount: 5,
    }
    component = shallow(<CampgroundTile { ...props } />);
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('CampgroundTile correctly indicates number of reviews', () => {
    expect(component.find('.CampgroundTile__review-count').text()).toBe('5 Reviews')
  });

  it('CampgroundTile correctly indicates number of reviews (when no reviews)', () => {
    props = { ratingCount: 0 }
    component = shallow(<CampgroundTile { ...props } />);
    expect(component.find('.CampgroundTile__review-count').text()).toBe('No reviews yet')
  });

  it('CampgroundTile component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
