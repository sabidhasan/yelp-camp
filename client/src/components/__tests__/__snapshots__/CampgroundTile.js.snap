// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CampgroundTile /> CampgroundTile component snapshot matches what is expected 1`] = `
<Fragment>
  <LazyLoad
    offset={0}
    once={true}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <img
      alt=""
      className="CampgroundTile__img"
    />
  </LazyLoad>
  <Link
    className="CampgroundTile__link"
    replace={false}
    to="campground/undefined"
  >
    <h2
      className="CampgroundTile__name bold"
    />
  </Link>
  <span
    className="CampgroundTile__region"
  />
  <RatingBar
    small={true}
  />
  <span
    className="CampgroundTile__review-count"
  >
    5 Reviews
  </span>
</Fragment>
`;
